vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/JunitRunner.java

```diff
@@ -55,27 +60,43 @@ public JunitRunner(OsgiFramework osgiFramework) {
         }
     }
 
-    public Bundle findTestBundle(String bundleNameSuffix) {
+    public String executeTests(TestDescriptor.TestCategory category, byte[] testConfig) {
+        testRuntimeProvider.initialize(testConfig);
+        Bundle testBundle = findTestBundle();
+        Optional<TestDescriptor> testDescriptor = loadTestDescriptor(testBundle);
+        if (testDescriptor.isEmpty()) {
+            throw new RuntimeException("Could not find test descriptor");
+        }
+        List<Class<?>> testClasses = loadClasses(testBundle, testDescriptor.get(), category);
+        return launchJunit(testClasses);
+    }
+
+    public boolean isSupported() {
+        Bundle testBundle = findTestBundle();
+        return loadTestDescriptor(testBundle).isPresent();
+    }
+
+    private Bundle findTestBundle() {
         return Stream.of(bundleContext.getBundles())
-                .filter(bundle -> bundle.getSymbolicName().endsWith(bundleNameSuffix))
+                .filter(bundle -> bundle.getSymbolicName().endsWith(TEST_BUNDLE_SUFFIX))
```

Comment:
Bundle symbolic name no longer contains `-tests` suffix. I will add it back if necessary.

Reply:
Iterate through all bundles and look for `X-JDisc-Test-Bundle-Version` header